pkg/apis/serving/v1alpha1/revision_validation.go

```diff
@@ -152,6 +155,34 @@ func (rs *RevisionSpec) Validate(ctx context.Context) *apis.FieldError {
 	return errs
 }
 
+func validateAnnotations(annotations map[string]string) *apis.FieldError {
+	return validatePercentageAnnotationKey(annotations, serving.QueueSideCarResourcePercentageAnnotation)
+}
+
+func validatePercentageAnnotationKey(annotations map[string]string, resourcePercentageAnnotationKey string) *apis.FieldError {
+	if len(annotations) == 0 {
+		return nil
+	}
+
+	v, ok := annotations[resourcePercentageAnnotationKey]
+	if !ok {
+		return nil
+	}
+	value, err := strconv.ParseFloat(v, 32)
+	if err != nil {
+		return &apis.FieldError{
+			Message: fmt.Sprintf("Invalid value %s for annotation %s: %v", v, resourcePercentageAnnotationKey, err),
+			Paths:   []string{resourcePercentageAnnotationKey},
+		}
```

Comment:
We have `apis.ErrInvalidValue`, but it is a bit awkward to properly use with map keys.  I think it should be something like: `apis.ErrInvalidValue(badValue, apis.CurrentField).ViaKey(resourcePercentageAnnotationKey)`

Then as it bubbles up it should properly show up as: `metadata.annotationsm[your.big/key]`